src/app.ts

```diff
@@ -75,18 +80,69 @@ async function app() {
       console.log('  -o, --out <out>');
       console.log('    Default: "fsh-generated"');
     })
-    .arguments('[path-to-fsh-defs]')
-    .action(function (pathToFshDefs) {
-      input = pathToFshDefs;
+    // NOTE: This option is included give a nice error message when the old init option is used while we support
+    // backwards compatibility of the build command.
+    .addOption(
+      new Option(
+        '-i, --init',
+        'ERROR: --init option is moved to a separate command. Run: sushi init'
+      ).hideHelp()
+    )
+    .on('option:init', () => {
+      // init was moved to a separate command, so log a message to indicate how to use it
+      console.log(
+        'The --init option has been moved to a separate command. Instead, run the following command: sushi init'
+      );
+      process.exit(1);
+    });
+
+  program
+    .command('init')
+    .description('initialize a SUSHI project')
+    .action(async function () {
+      await init().catch(logUnexpectedError);
+      process.exit(0);
+    });
+
+  program
+    .command('update-dependencies')
+    .description('update FHIR packages in project configuration')
+    .argument('[path-to-fsh-project]')
+    .action(async function (projectPath) {
+      await runUpdateDependencies(projectPath).catch(logUnexpectedError);
+      process.exit(0);
     })
-    .showHelpAfterError()
-    .parse(process.argv)
-    .opts();
+    .on('--help', () => {
+      console.log('');
+      console.log('Additional information:');
+      console.log('  [path-to-fsh-project]');
+      console.log('    Default: "."');
+    });
+
+  program.parse(process.argv).opts();
+}
 
-  if (program.init) {
-    await init();
-    process.exit(0);
+async function runUpdateDependencies(projectPath: string) {
+  const input = ensureInputDir(projectPath);
+  const config: Configuration = readConfig(input);
+  await updateExternalDependencies(config);
+}
+
+async function runBuild(input: string, program: OptionValues, helpText: string) {
+  // NOTE: This is included to provide nicer handling for the previous CLI structure for building FSH projects
+  // Before doing anything else, we do our best to decide if the legacy SUSHI command structure was used
+  // (i.e. sushi . or sushi -d . or sushi -d or sushi)
+  const isLegacyBuildCommand =
+    !process.argv.includes('build') &&
+    (process.argv.slice(2).some(a => fs.existsSync(a)) ||
+      process.argv.slice(2).every(a => a.startsWith('-')));
+  if (!isLegacyBuildCommand) {
+    // This was probably an error (maybe a typo of a real command),
+    // not someone trying to use the old syntax, so exit
+    console.log(helpText);
+    process.exit(1);
   }
```

Comment:
Unfortunately, there are a few issues with this:
1. I think this would still fail for a command like `sushi -o /my/output/dir` when `/my/output/dir` doesn't exist (even though SUSHI can create it).
2. This would definitely fail if someone invoked SUSHI the recommended way (`sushi build .`) because `isLegacyBuildCommand` would be false (and rightly so), which would then cause us to exit with an error due to the condition on line 139.  Oops!
3. I think that the command _has_ to be the first argument to SUSHI. If I'm correct about that, then I think we can simplify this?

```suggestion
    // Check the first argument passed into sushi. If it is not "build", then this is a legacy build,
  // in which case we should make sure that the first argument is a flag or a valid path.
  // NOTE: The index of the argument of interest differs depending on how sushi is invoked, e.g.,
  // argv[1] for `sushi build /some/dir`, but argv[2] for `ts-node src/app.ts build /some/dir`.
  const arg = /(ts-)?node(\.[a-zA-Z]+)?/.test(process.argv[0]) ? process.argv[2] : process.argv[1];
  if (arg != null && arg !== 'build' && !arg.startsWith('-') && !fs.existsSync(arg)) {
    // It's not a flag or a path, so it's probably a typo of an existing command
    console.log(helpText);
    process.exit(1);
  }
```
(Although I guess you might argue it's not as simple now that we have to check for `node`/`ts-node`.  What do you think?)

Reply:
Some additional thoughts, but I think this is a good change!

1. Ah, I had forgotten we actually create it, even if it doesn't exist.
2. I thought `sushi build .` would work because the first part of `isLegacyBuildCommand` check if `'build` was not included. That still doesn't fix 1. though.
3. I think your simplified approach makes sense! But now I'm wondering if we can simplify it further. Is there ever a case where we do want to use `process.argv[1]`? When I tried to `npm pack` and use the packed version, the first argument is still a `node` argument, so I think it might be safe to just always have `const arg = process.argv[2]`.

Reply:
> I thought `sushi build .` would work because the first part of `isLegacyBuildCommand` check if `build` was not included. That still doesn't fix 1. though.

Right, the first part of `isLegacyBuildCommand` is `!process.argv.includes('build')`.  So, if the args include `build`, then that evaluates to `false` -- and since it's `&&`ed w/ the rest, `isLegacyBuildCommand` becomes `false`.  Then line 139 is `if (!isLegacyBuildCommand)` -- so when `build` is included in the args, that becomes `if (!false)`, which is basically `if (true)`.  And what do we do `if (true)`? We `process.exit(1)`.

> When I tried to npm pack and use the packed version, the first argument is still a node argument, so I think it might be safe to just always have const arg = process.argv[2].

Oops.  You're probably right.  I likely made a bad assumption there!